File: software/examples/go/example_simple.go
package main

import (
	"fmt"
	"tinkerforge/ipconnection"
	"tinkerforge/load_cell_bricklet"
)

const ADDR string = "localhost:4223"
const UID string = "XYZ" // Change XYZ to the UID of your Load Cell Bricklet.

func main() {
	ipcon := ipconnection.New()
	defer ipcon.Close()
	lc, _ := load_cell_bricklet.New(UID, &ipcon) // Create device object.

	ipcon.Connect(ADDR) // Connect to brickd.
	defer ipcon.Disconnect()
	// Don't use device before ipcon is connected.

	// Get current weight.
	weight, _ := lc.GetWeight()
	fmt.Printf("Weight:  g\n", weight)

	fmt.Print("Press enter to exit.")
	fmt.Scanln()

}

File: software/examples/rust/example_simple.rs
use std::{io, error::Error};

use tinkerforge::{ip_connection::IpConnection, 
                  load_cell_bricklet::*};


const HOST: &str = "localhost";
const PORT: u16 = 4223;
const UID: &str = "XYZ"; // Change XYZ to the UID of your Load Cell Bricklet.

fn main() -> Result<(), Box<dyn Error>> {
    let ipcon = IpConnection::new(); // Create IP connection.
    let lc = LoadCellBricklet::new(UID, &ipcon); // Create device object.

    ipcon.connect((HOST, PORT)).recv()??; // Connect to brickd.
    // Don't use device before ipcon is connected.

		// Get current weight.
let weight = lc.get_weight().recv()?;
		println!("Weight: {} g", weight);

    println!("Press enter to exit.");
    let mut _input = String::new();
    io::stdin().read_line(&mut _input)?;
    ipcon.disconnect();
    Ok(())
}
